assets/shaders/rally_point.wgsl

```diff
@@ -0,0 +1,35 @@
+#import bevy_pbr::mesh_view_bindings
+#import bevy_pbr::mesh_bindings
+
+struct CustomMaterial {
+    color: vec4<f32>,
+    pointiness: f32,
+    speed: f32,
+    length: f32,
+    spacing: f32,
+    fade: f32,
+};
+
+@group(1) @binding(0)
+var<uniform> material: CustomMaterial;
+
+@fragment
+fn fragment(
+    #import bevy_pbr::mesh_vertex_output
+) -> @location(0) vec4<f32> {
+    let world_space_length: f32 = length(mesh.model[0].xyz);
+    let scaled_x: f32 = uv.x * world_space_length;
+    let offset_y: f32 = abs(uv.y - 0.5) * material.pointiness;
+    let scaled_time: f32 = globals.time * material.speed;
+    let total_length = material.length + material.spacing;
+
+    let value = scaled_x + offset_y - scaled_time;
+    let clamped_value = (value % total_length + total_length) % total_length;
```

Comment:
What's the difference between `(value % total_length + total_length) % total_length;` and `value % total_length;`?

Reply:
In wgsl `(-3) % 7` gives -3, when I wanted it to be +4.

Reply:
I see, I didn't realize that. Could you add a code comment explaining that?